configure.ac

```diff
@@ -575,13 +575,25 @@ AS_IF([test "x$with_gc" = xboehm],
   ]
 )
 
+dnl
+dnl Define kernel version
+dnl
+
+
+gap_kernel_major_version=4
+gap_kernel_minor_version=0
+AC_SUBST([gap_kernel_major_version])
+AC_SUBST([gap_kernel_minor_version])
+
+AC_DEFINE_UNQUOTED([GAP_KERNEL_MAJOR_VERSION], $gap_kernel_major_version, [The major version of the kernel ABI])
+AC_DEFINE_UNQUOTED([GAP_KERNEL_MINOR_VERSION], $gap_kernel_minor_version, [The minor version of the kernel ABI])
 
 dnl
 dnl User setting: Compatibility mode (on by default)
 dnl
 AS_IF([test "x$enable_hpcgap" = xyes],
-  [GAPARCH="$host-hpcgap${ABI}"],
-  [GAPARCH="$host-default${ABI}"])
+  [GAPARCH="$host-hpcgap${ABI}-abi${gap_kernel_major_version}"],
+  [GAPARCH="$host-default${ABI}-abi${gap_kernel_major_version}"])
```

Comment:
That seems like a good idea to me. Although the "ABI" vs. "abi" bit are a bit confusing, and I worry that down the road it'll lead to confusion in discussions. Perhaps we should rename `ABI` to `ARCHABI`. And/or change `-abi` to `-kernel`, `-kv`, etc.?

Reply:
I like `-kv`. I didn't want something that looks too much like "version", because then people might expect it to be "4.9.2" or something like that

Reply:
Just to clarify: that's an optional remark, i.e., it could be changed now, or later, or never